Sources/Swift/Protocol/Metric.swift

```diff
@@ -0,0 +1,289 @@
+/// A metric entry that captures metric data with associated attribute metadata.
+///
+/// Use the `options.beforeSendMetric` callback to modify or filter metric data.
+public struct Metric {
```

Comment:
`m`: Should we maybe add a comment explaining that this is Swift only by intention, and that we still figuring out wheter or how this should be exposed to Objc?

Reply:
I don't think there's any value in that comment, because if we add it here, Objective-C users won't see it because they don't have access to the Swift type. And if a Swift user sees it, they are not the target group of the comment. I have already added a note to our documentation draft PR: https://github.com/getsentry/sentry-docs/pull/15822